posts/intro-to-mathport.md

```diff
@@ -0,0 +1,224 @@
+  This is what lets us do things like:
+```
+import Mathbin
+
+#lookup3 algebraic_geometry.Scheme
+#check AlgebraicGeometry.Scheme
+```
+  Yay, Lean 4 has schemes! :-)
+  To see this file in action, you should check out a copy of the `mathlib3port` repository described below,
+  and make a new file there.
+* `synport` constructs Lean 4 `.lean` files, on a "best effort" basis.
+  (It uses both the output from binport, and Lean 3's `--ast` output to guide it.)
+  We should not expect that this will ever converge to a perfect translator.
+  Instead the hope is that it gives us something that humans can plausibly improve to a complete translation of `mathlib3`.
+
+To understand how `mathport` (mostly talking about the synport part from here on) works,
+it's important to understand that it is translating `mathlib3` to Lean 4 source code, "modulo" the current content of `mathlib4`.
+That is, the premise is that as we progressively construct `mathlib4`
+(whether by translating by hand, moving content from `mathport`'s output to `mathlib4`, or adding `#align` statements)
+the output from running `mathport` on `mathlib3` will change.
+In particular, as `mathport` is translating each declaration,
+it checks to see if a corresponding declaration in `mathlib4` already exists, and is defeq.
+If so, `mathport` will instead just use that declaration.
+If not, `mathport` will make a copy, appending a `×` to the name.
+Sometimes these misalignments are due to an unintentional non-defeq, that can be fixed in `mathlib4`.
+Other times, we genuinely want to change something in `mathlib4`
+(e.g. to use Lean 4's multiple parent structures, which are better than `old_structure_cmd`).
+As a result, we expect that some misalignments will persist throughout the `mathport`-assisted stage of the port,
+and only afterwards will we polish these away.
+
+A detailed account of how binport and synport are working is beyond the scope of this blog post.
+Hopefully we'll have one eventually,
+but in the meantime [Mario's talk](https://www.youtube.com/watch?v=fOO2fByx8tw) is very helpful.
+
+# What should I look at?
+
+Please note that `mathport` takes considerable resources to run on `mathlib3`: approximately 3.5 hours, and 32gb of RAM.
+So you'll probably want to look at artifacts generated by CI rather than running it yourself.
+
+* https://github.com/leanprover-community/mathlib3port should be most people's first stop.
+  This contains a copy of the `.lean` files produced by a recent run of `mathport`, in the `Mathbin` directory.
+  You should just be able to check out a copy of this repository, and open the folder in VS Code,
+  to see the current state of `mathport` output.
+  You can also try out the above example with `#check AlgebraicGeometry.Scheme` in a fresh file here.
+  * Remember it's expected to be horribly broken; most tactics aren't implemented,
+    and there are bugs around parenthesization of arguments, and name resolution!
```

Comment:
In this item, can you explicitly say that the repository contains synported `.lean` files, and gives you easy access to binported `.olean` files? It would also be clearer if you are a bit more specific when `mathport` means `synport` and when it means `binport`. Something like
```suggestion
* https://github.com/leanprover-community/mathlib3port should be most people's first stop.
  This contains a copy of the `.lean` files produced by a recent run of `synport`, in the `Mathbin` directory.
  Furthermore, you can use `lake build` to download the `.olean` files generated by `binport`.
  You should just be able to check out a copy of this repository, and open the folder in VS Code,
  to see the current state of `mathport` output.
  You can also try out the above example with `#check AlgebraicGeometry.Scheme` in a fresh file here.
  * Remember the synported files are expected to be horribly broken; most tactics aren't implemented,
    and there are bugs around parenthesization of arguments, and name resolution!
```